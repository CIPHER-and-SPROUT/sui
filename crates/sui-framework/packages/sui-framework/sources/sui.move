// Copyright (c) Mysten Labs, Inc.
// SPDX-License-Identifier: Apache-2.0

/// Coin<SUI> is the token used to pay for gas in Sui.
/// It has 9 decimals, and the smallest unit (10^-9) is called "mist".
module sui::sui {
    use std::option;
    use sui::tx_context::{Self, TxContext};
    use sui::balance::{Self, Balance};
    use sui::transfer;
    use sui::coin;

    const EAlreadyMinted: u64 = 0;
    /// Sender is not @0x0 the system address.
    const ENotSystemAddress: u64 = 1;

    /// The amount of Mist per Sui token based on the the fact that mist is
    /// 10^-9 of a Sui token
    const MIST_PER_SUI: u64 = 1_000_000_000;

    /// The total supply of Sui denominated in whole Sui tokens (10 Billion)
    const TOTAL_SUPPLY_SUI: u64 = 10_000_000_000;

    /// The total supply of Sui denominated in Mist (10 Billion * 10^9)
    const TOTAL_SUPPLY_MIST: u64 = 10_000_000_000_000_000_000;

    /// Name of the coin
    struct SUI has drop {}

    #[allow(unused_function)]
    /// Register the `SUI` Coin to acquire its `Supply`.
    /// This should be called only once during genesis creation.
    fun new(ctx: &mut TxContext): Balance<SUI> {
        assert!(tx_context::sender(ctx) == @0x0, ENotSystemAddress);
        assert!(tx_context::epoch(ctx) == 0, EAlreadyMinted);

        let (treasury, metadata) = coin::create_currency(
            SUI {},
            9,
            b"SUI",
            b"Sui",
            // TODO: add appropriate description and logo url
            b"",
            option::none(),
            ctx
        );
        transfer::public_freeze_object(metadata);
        let supply = coin::treasury_into_supply(treasury);
        let total_sui = balance::increase_supply(&mut supply, TOTAL_SUPPLY_MIST);
        balance::destroy_supply(supply);
        total_sui
    }

    public entry fun transfer(c: coin::Coin<SUI>, recipient: address) {
        transfer::public_transfer(c, recipient)
    }

    #[test_only]    
    /// Initiates the Sui Coin in tests.
    /// It allows the `sui::coin::CoinMetadata` to be accessible in test modules. 
    /// The Sui Coin decimals is often needed in DeFi to perform fixed-point math operations.  
    /// The functions `sui::coin::mint_for_testing` and `sui::balance::create_for_testing` allow 
    /// the user to mint Sui but does not share its metadata.
    public fun new_for_testing(ctx: &mut TxContext): Balance<SUI> {
        new(ctx)
    }
}
