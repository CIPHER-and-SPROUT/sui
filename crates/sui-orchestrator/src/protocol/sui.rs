// Copyright (c) Mysten Labs, Inc.
// SPDX-License-Identifier: Apache-2.0

use std::path::PathBuf;

use sui_config::genesis_config::{GenesisConfig, ValidatorGenesisInfo};
use sui_types::multiaddr::Multiaddr;

use crate::{benchmark::BenchmarkParameters, client::Instance, settings::Settings};

use super::{ProtocolCommands, ProtocolMetrics};

/// All configurations information to run a sui client or validator.
pub struct SuiProtocol {
    /// The working directory on the remote hosts (containing the databases and configuration files).
    working_dir: PathBuf,
}

impl ProtocolCommands for SuiProtocol {
    const NODE_METRICS_PORT: u16 =
        ValidatorGenesisInfo::DEFAULT_METRICS_PORT + GenesisConfig::BENCHMARKS_PORT_OFFSET as u16;
    const CLIENT_METRICS_PORT: u16 = 8081;

    fn protocol_dependencies() -> Vec<&'static str> {
        vec![
            // Install typical sui dependencies.
            "sudo apt-get -y install curl git-all clang cmake gcc libssl-dev pkg-config libclang-dev",
            // This dependency is missing from the Sui docs.
            "sudo apt-get -y install libpq-dev",
        ]
    }

    fn db_directories(&self) -> Vec<PathBuf> {
        let authorities_db = [&self.working_dir, &sui_config::AUTHORITIES_DB_NAME.into()]
            .iter()
            .collect();
        let consensus_db = [&self.working_dir, &sui_config::CONSENSUS_DB_NAME.into()]
            .iter()
            .collect();
        vec![authorities_db, consensus_db]
    }

    fn genesis_command<'a, I>(&self, instances: I) -> String
    where
        I: Iterator<Item = &'a Instance>,
    {
        let working_dir = self.working_dir.display();
        let ips = instances
            .map(|x| x.main_ip.to_string())
            .collect::<Vec<_>>()
            .join(" ");
        let genesis = [
            "cargo run --release --bin sui --",
            "genesis",
            &format!("-f --working-dir {working_dir} --benchmark-ips {ips}"),
        ]
        .join(" ");

        [
            &format!("mkdir -p {working_dir}"),
            "source $HOME/.cargo/env",
            &genesis,
        ]
        .join(" && ")
    }

    fn node_command<'a, I>(&self, instances: I) -> Box<dyn Fn(usize) -> String>
    where
        I: Iterator<Item = &'a Instance>,
    {
        let instances: Vec<_> = instances.cloned().collect();
        let listen_addresses = Self::make_listen_addresses(&instances);

        let working_dir = self.working_dir.clone();
        Box::new(move |i| {
            let validator_config = sui_config::validator_config_file(i);
            let config_path: PathBuf = [&working_dir, &validator_config.into()].iter().collect();
            let path = config_path.display();
            let address = listen_addresses[i].clone();

            let run = [
                "cargo run --release --bin sui-node --",
                &format!("--config-path {path} --listen-address {address}"),
            ]
            .join(" ");
            ["source $HOME/.cargo/env", &run].join(" && ")
        })
    }

    fn client_command<'a, I>(
        &self,
        _instances: I,
        parameters: &BenchmarkParameters,
    ) -> Box<dyn Fn(usize) -> String>
    where
        I: Iterator<Item = &'a Instance>,
    {
        let genesis_path: PathBuf = [&self.working_dir, &sui_config::SUI_GENESIS_FILENAME.into()]
            .iter()
            .collect();
        let keystore_path: PathBuf = [
            &self.working_dir,
            &sui_config::SUI_BENCHMARK_GENESIS_GAS_KEYSTORE_FILENAME.into(),
        ]
        .iter()
        .collect();
        let committee_size = parameters.nodes;
        let load_share = parameters.load / committee_size;
        let shared_counter = parameters.shared_objects_ratio;
        let transfer_objects = 100 - shared_counter;
        let metrics_port = Self::CLIENT_METRICS_PORT;

        Box::new(move |i| {
            let genesis = genesis_path.display();
            let keystore = keystore_path.display();
            // let gas_id = SuiProtocol::gas_object_id_offsets(committee_size)[i].clone();
            let gas_id = GenesisConfig::benchmark_gas_object_id_offsets(committee_size)[i].clone();
            let run = [
                "cargo run --release --bin stress --",
                "--num-client-threads 24 --num-server-threads 1",
                "--local false --num-transfer-accounts 2",
                &format!("--genesis-blob-path {genesis} --keystore-path {keystore}",),
                &format!("--primary-gas-id {gas_id}"),
                "bench",
                &format!("--in-flight-ratio 30 --num-workers 24 --target-qps {load_share}"),
                &format!("--shared-counter {shared_counter} --transfer-object {transfer_objects}"),
                &format!("--client-metric-host 0.0.0.0 --client-metric-port {metrics_port}"),
            ]
            .join(" ");
            ["source $HOME/.cargo/env", &run].join(" && ")
        })
    }
}

impl ProtocolMetrics for SuiProtocol {
    const BENCHMARK_DURATION: &'static str = "benchmark_duration";
    const TOTAL_TRANSACTIONS: &'static str = "latency_s_count";
    const LATENCY_BUCKETS: &'static str = "latency_s";
    const LATENCY_SUM: &'static str = "latency_s_sum";
    const LATENCY_SQUARED_SUM: &'static str = "latency_squared_s";
}

impl SuiProtocol {
    /// Make a new instance of the Sui protocol commands generator.
    pub fn new(settings: &Settings) -> Self {
        Self {
            working_dir: [&settings.working_dir, &sui_config::SUI_CONFIG_DIR.into()]
                .iter()
                .collect(),
        }
    }

    /// Convert the ip of the validators' network addresses to 0.0.0.0.
    pub fn make_listen_addresses(instances: &[Instance]) -> Vec<Multiaddr> {
        let ips: Vec<_> = instances.iter().map(|x| x.main_ip.to_string()).collect();
        let genesis_config = GenesisConfig::new_for_benchmarks(&ips);
        let mut addresses = Vec::new();
        if let Some(validator_configs) = genesis_config.validator_config_info.as_ref() {
            for validator_info in validator_configs {
                let address = &validator_info.genesis_info.network_address;
                addresses.push(address.zero_ip_multi_address());
            }
        }
        addresses
    }
}
