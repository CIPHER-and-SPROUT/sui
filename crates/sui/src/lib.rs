// Copyright (c) 2021, Facebook, Inc. and its affiliates
// Copyright (c) Mysten Labs, Inc.
// SPDX-License-Identifier: Apache-2.0

pub mod client_commands;
pub mod console;
pub mod fire_drill;
pub mod keytool;
pub mod package_cache;
pub mod package_hooks;
pub mod shell;
pub mod sui_commands;
pub mod validator_commands;
pub mod zklogin_commands_util;

pub mod genesis_ceremony;
pub mod genesis_inspector;
