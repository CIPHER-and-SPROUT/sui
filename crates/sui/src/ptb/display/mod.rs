// Copyright (c) Mysten Labs, Inc.
// SPDX-License-Identifier: Apache-2.0

mod gas_cost_summary;
mod status;
mod summary;

use crate::ptb::ptb::PTBGas;
use crate::ptb::ptb::PTBPreview;
use std::fmt::{Display, Formatter};
use tabled::{
    builder::Builder as TableBuilder,
    settings::{style::HorizontalLine, Panel as TablePanel, Style as TableStyle},
};

pub struct Pretty<'a, T>(pub &'a T);

impl Display for PTBPreview {
    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
        let mut builder = TableBuilder::default();
        let columns = vec!["command", "from", "value(s)"];
        builder.set_header(columns);
        let mut from = "console";
        for cmd in &self.cmds {
            if cmd.name == "file-include-start" {
                from = cmd.values.first().unwrap();
                continue;
            } else if cmd.name == "file-include-end" {
                from = "console";
                continue;
            } else if cmd.is_preview_false() || cmd.is_warn_shadows_false() {
                continue;
            }
            builder.push_record([
                cmd.name.to_string(),
                from.to_string(),
                cmd.values.join(" ").to_string(),
            ]);
        }
        let mut table = builder.build();
        table.with(TablePanel::header("PTB Preview"));
        table.with(TableStyle::rounded().horizontals([
            HorizontalLine::new(1, TableStyle::modern().get_horizontal()),
            HorizontalLine::new(2, TableStyle::modern().get_horizontal()),
            HorizontalLine::new(2, TableStyle::modern().get_horizontal()),
        ]));
        table.with(tabled::settings::style::BorderSpanCorrection);

        write!(f, "{}", table)
    }
}

impl Display for PTBGas {
    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
        let r = match self {
            PTBGas::Min => "min",
            PTBGas::Max => "max",
            PTBGas::Sum => "sum",
        };
        write!(f, "{}", r)
    }
}
