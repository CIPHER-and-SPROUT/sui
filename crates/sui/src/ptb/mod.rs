// Copyright (c) Mysten Labs, Inc.
// SPDX-License-Identifier: Apache-2.0

pub mod display;
pub mod ptb;
pub mod ptb_builder;
