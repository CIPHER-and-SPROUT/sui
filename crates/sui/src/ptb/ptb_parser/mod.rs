// Copyright (c) Mysten Labs, Inc.
// SPDX-License-Identifier: Apache-2.0

pub mod argument;
pub mod argument_token;
pub mod build_ptb;
pub mod command_token;
pub mod context;
pub mod errors;
pub mod parser;
pub mod utils;
