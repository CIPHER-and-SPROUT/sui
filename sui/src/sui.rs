// Copyright (c) 2022, Mysten Labs, Inc.
// SPDX-License-Identifier: Apache-2.0
extern crate core;

use clap::*;
use sui::sui_commands::SuiCommand;

#[cfg(test)]
#[path = "unit_tests/cli_tests.rs"]
mod cli_tests;

#[derive(Parser)]
#[clap(
    name = "Sui Local",
    about = "A Byzantine fault tolerant chain with low-latency finality and high throughput",
    rename_all = "kebab-case"
)]
struct SuiOpt {
    #[clap(subcommand)]
    command: SuiCommand,
}

#[tokio::main]
async fn main() -> Result<(), anyhow::Error> {
    let config = telemetry_subscribers::TelemetryConfig {
        service_name: "sui".into(),
        enable_tracing: std::env::var("SUI_TRACING_ENABLE").is_ok(),
        json_log_output: std::env::var("SUI_JSON_SPAN_LOGS").is_ok(),
        ..Default::default()
    };
    #[allow(unused)]
    let guard = telemetry_subscribers::init(config);

    let options: SuiOpt = SuiOpt::parse();
    options.command.execute().await
}
