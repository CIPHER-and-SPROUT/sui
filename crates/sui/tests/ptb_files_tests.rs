// Copyright (c) Mysten Labs, Inc.
// SPDX-License-Identifier: Apache-2.0

use clap::CommandFactory;
use std::{collections::BTreeMap, path::Path};
use sui::ptb::{
    ptb::PTB,
    ptb_parser::{errors::render_errors, parser::PTBParser},
};
use test_cluster::TestClusterBuilder;

// check if all good ptb files can be parsed OK
fn test_parse_all_ptb_files(path: &Path) -> datatest_stable::Result<()> {
    let ptb = PTB::default();
    let cmd = PTB::command();
    let file = path.to_str().unwrap();
    let args = cmd.get_matches_from(vec!["ptb", "--file", file]);
    let cwd = std::env::current_dir().unwrap();
    let commands = ptb.from_matches(cwd, &args, &mut BTreeMap::new()).unwrap();

    let mut parser = PTBParser::new();
    for (_, cmd) in commands {
        parser.parse(cmd.clone());
    }
    let (_parsed_commands, errors) = parser.finish();
    assert!(errors.is_empty());

    Ok(())
}

// check if the bad ptb files return errors when parsed
fn test_parse_bad_ptb_files(path: &Path) -> datatest_stable::Result<()> {
    let ptb = PTB::default();
    let cmd = PTB::command();
    let file = path.to_str().unwrap();
    let args = cmd.get_matches_from(vec!["ptb", "--file", file]);
    let cwd = std::env::current_dir().unwrap();
    let commands = ptb.from_matches(cwd, &args, &mut BTreeMap::new()).unwrap();
    let mut parser = PTBParser::new();
    for (_, cmd) in commands.clone() {
        parser.parse(cmd.clone());
    }
    let (_parsed_commands, errors) = parser.finish();
    let rendered = render_errors(commands, errors.clone());
    for e in rendered.iter() {
        println!("{:?}", e);
    }
    println!("{:?}", errors.len());
    assert!(!errors.is_empty());
    Ok(())
}

// parse, build PTB, and run it locally in a network
async fn test_build_and_run_ptb_from_files(path: &Path) -> datatest_stable::Result<()> {
    let ptb = PTB::default();
    let cmd = PTB::command();
    let file = path.to_str().unwrap();
    let args = cmd.get_matches_from(vec!["ptb", "--file", file]);
    let test_cluster = TestClusterBuilder::new().build().await;
    let context = test_cluster.wallet;

    ptb.execute(args, Some(context)).await.unwrap();
    Ok(())
}

async fn test_no_gas_picker_from_files(path: &Path) -> datatest_stable::Result<()> {
    let ptb = PTB::default();
    let cmd = PTB::command();
    let file = path.to_str().unwrap();
    let args = cmd.get_matches_from(vec!["ptb", "--file", file]);
    let test_cluster = TestClusterBuilder::new().build().await;
    let context = test_cluster.wallet;

    let execute = ptb.execute(args, Some(context)).await;
    assert!(execute.is_err());
    Ok(())
}

datatest_stable::harness!(
    test_parse_all_ptb_files,
    "tests/ptb_files",
    r"^*.ptb$",
    test_parse_bad_ptb_files,
    "tests/ptb_files",
    r"^*.bad",
    // test_build_and_run_ptb_from_files,
    // "tests/ptb_files",
    // r"^run_*.sh",
    // test_no_gas_picker_from_files
    // "tests/ptb_files",
    // "no_gas_picker",
);
