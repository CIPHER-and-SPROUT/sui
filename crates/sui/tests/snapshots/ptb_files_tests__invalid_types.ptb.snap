---
source: crates/sui/tests/ptb_files_tests.rs
expression: "results.join(\"\\n\")"
---
 === PREVIEW === 
╭──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│ PTB Preview                                                                                                                                          │
├───────────────┬───────────────────────────────────────────┬──────────────────────────────────────────────────────────────────────────────────────────┤
│ command       │ from                                      │ value(s)                                                                                 │
├───────────────┼───────────────────────────────────────────┼──────────────────────────────────────────────────────────────────────────────────────────┤
│ assign        │ tests/ptb_files/parsing/invalid_types.ptb │ a 0xfffu8                                                                                │
│ assign        │ tests/ptb_files/parsing/invalid_types.ptb │ b 0xffffffffu16                                                                          │
│ assign        │ tests/ptb_files/parsing/invalid_types.ptb │ c 0xfffffffffffffu32                                                                     │
│ assign        │ tests/ptb_files/parsing/invalid_types.ptb │ d 0xffffffffffffffffffffffffffffffu64                                                    │
│ assign        │ tests/ptb_files/parsing/invalid_types.ptb │ e 0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffu128                       │
│ assign        │ tests/ptb_files/parsing/invalid_types.ptb │ f 0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffu256 │
│ make_move_vec │ tests/ptb_files/parsing/invalid_types.ptb │ <string> ['foo']                                                                         │
│ make_move_vec │ tests/ptb_files/parsing/invalid_types.ptb │ <String> ['foo']                                                                         │
│ make_move_vec │ tests/ptb_files/parsing/invalid_types.ptb │ <Bool> [true, false]                                                                     │
│ make_move_vec │ tests/ptb_files/parsing/invalid_types.ptb │ <i8> [0]                                                                                 │
│ make_move_vec │ tests/ptb_files/parsing/invalid_types.ptb │ <i16> [0]                                                                                │
│ make_move_vec │ tests/ptb_files/parsing/invalid_types.ptb │ <i32> [0]                                                                                │
│ make_move_vec │ tests/ptb_files/parsing/invalid_types.ptb │ <i64> [0]                                                                                │
│ make_move_vec │ tests/ptb_files/parsing/invalid_types.ptb │ <i128> [0]                                                                               │
│ make_move_vec │ tests/ptb_files/parsing/invalid_types.ptb │ <i256> [0]                                                                               │
│ make_move_vec │ tests/ptb_files/parsing/invalid_types.ptb │ <Vector<u8>> [vector[0]]                                                                 │
│ make_move_vec │ tests/ptb_files/parsing/invalid_types.ptb │ <vector<U256>> [vector[0]]                                                               │
│ make_move_vec │ tests/ptb_files/parsing/invalid_types.ptb │ <vector<i256>> [vector[0]]                                                               │
│ preview       │ console                                   │ true                                                                                     │
╰───────────────┴───────────────────────────────────────────┴──────────────────────────────────────────────────────────────────────────────────────────╯
 === ERRORS AFTER PARSING INPUT COMMANDS === 
  × Error at command 0 in PTB file 'tests/ptb_files/parsing/invalid_types.ptb'
   ╭────
 1 │ assign a 0xfffu8 
   ·          ───┬───
   ·             ╰── Failed to parse arguments for 'assign' command. number too large to fit in target type
   ╰────

  × Error at command 1 in PTB file 'tests/ptb_files/parsing/invalid_types.ptb'
   ╭────
 1 │ assign b 0xffffffffu16 
   ·          ──────┬──────
   ·                ╰── Failed to parse arguments for 'assign' command. number too large to fit in target type
   ╰────

  × Error at command 2 in PTB file 'tests/ptb_files/parsing/invalid_types.ptb'
   ╭────
 1 │ assign c 0xfffffffffffffu32 
   ·          ─────────┬────────
   ·                   ╰── Failed to parse arguments for 'assign' command. number too large to fit in target type
   ╰────

  × Error at command 3 in PTB file 'tests/ptb_files/parsing/invalid_types.ptb'
   ╭────
 1 │ assign d 0xffffffffffffffffffffffffffffffu64 
   ·          ─────────────────┬─────────────────
   ·                           ╰── Failed to parse arguments for 'assign' command. number too large to fit in target type
   ╰────

  × Error at command 4 in PTB file 'tests/ptb_files/parsing/invalid_types.ptb'
   ╭────
 1 │ assign e 0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffu128 
   ·          ────────────────────────────────┬───────────────────────────────
   ·                                          ╰── Failed to parse arguments for 'assign' command. number too large to fit in target type
   ╰────

  × Error at command 5 in PTB file 'tests/ptb_files/parsing/invalid_types.ptb'
   ╭────
 1 │ assign f 0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffu256 
   ·          ───────────────────────────────────────────┬──────────────────────────────────────────
   ·                                                     ╰── Failed to parse arguments for 'assign' command. Invalid string length
   ╰────

  × Error at command 6 in PTB file 'tests/ptb_files/parsing/invalid_types.ptb'
   ╭────
 1 │ make_move_vec <string> ['foo'] 
   ·               ────┬───
   ·                   ╰── Failed to parse arguments for 'make_move_vec' command. expected token ::, got >
   ╰────

  × Error at command 7 in PTB file 'tests/ptb_files/parsing/invalid_types.ptb'
   ╭────
 1 │ make_move_vec <String> ['foo'] 
   ·               ────┬───
   ·                   ╰── Failed to parse arguments for 'make_move_vec' command. expected token ::, got >
   ╰────

  × Error at command 8 in PTB file 'tests/ptb_files/parsing/invalid_types.ptb'
   ╭────
 1 │ make_move_vec <Bool> [true, false] 
   ·               ───┬──
   ·                  ╰── Failed to parse arguments for 'make_move_vec' command. expected token ::, got >
   ╰────

  × Error at command 9 in PTB file 'tests/ptb_files/parsing/invalid_types.ptb'
   ╭────
 1 │ make_move_vec <i8> [0] 
   ·               ──┬─
   ·                 ╰── Failed to parse arguments for 'make_move_vec' command. expected token ::, got >
   ╰────

  × Error at command 10 in PTB file 'tests/ptb_files/parsing/
  │ invalid_types.ptb'
   ╭────
 1 │ make_move_vec <i16> [0] 
   ·               ──┬──
   ·                 ╰── Failed to parse arguments for 'make_move_vec' command. expected token ::, got >
   ╰────

  × Error at command 11 in PTB file 'tests/ptb_files/parsing/
  │ invalid_types.ptb'
   ╭────
 1 │ make_move_vec <i32> [0] 
   ·               ──┬──
   ·                 ╰── Failed to parse arguments for 'make_move_vec' command. expected token ::, got >
   ╰────

  × Error at command 12 in PTB file 'tests/ptb_files/parsing/
  │ invalid_types.ptb'
   ╭────
 1 │ make_move_vec <i64> [0] 
   ·               ──┬──
   ·                 ╰── Failed to parse arguments for 'make_move_vec' command. expected token ::, got >
   ╰────

  × Error at command 13 in PTB file 'tests/ptb_files/parsing/
  │ invalid_types.ptb'
   ╭────
 1 │ make_move_vec <i128> [0] 
   ·               ───┬──
   ·                  ╰── Failed to parse arguments for 'make_move_vec' command. expected token ::, got >
   ╰────

  × Error at command 14 in PTB file 'tests/ptb_files/parsing/
  │ invalid_types.ptb'
   ╭────
 1 │ make_move_vec <i256> [0] 
   ·               ───┬──
   ·                  ╰── Failed to parse arguments for 'make_move_vec' command. expected token ::, got >
   ╰────

  × Error at command 15 in PTB file 'tests/ptb_files/parsing/
  │ invalid_types.ptb'
   ╭────
 1 │ make_move_vec <Vector<u8>> [vector[0]] 
   ·               ──────┬─────
   ·                     ╰── Failed to parse arguments for 'make_move_vec' command. expected token ::, got <
   ╰────

  × Error at command 16 in PTB file 'tests/ptb_files/parsing/
  │ invalid_types.ptb'
   ╭────
 1 │ make_move_vec <vector<U256>> [vector[0]] 
   ·               ───────┬──────
   ·                      ╰── Failed to parse arguments for 'make_move_vec' command. expected token ::, got >
   ╰────

  × Error at command 17 in PTB file 'tests/ptb_files/parsing/
  │ invalid_types.ptb'
   ╭────
 1 │ make_move_vec <vector<i256>> [vector[0]] 
   ·               ───────┬──────
   ·                      ╰── Failed to parse arguments for 'make_move_vec' command. expected token ::, got >
   ╰────

